### PR TITLE
Add fixes to async docs

### LINKED ISSUE #157: The "#" in headers for the F# async doc is missing
The headers are missing "#".  Strangely, [the raw markdown](https://raw.githubusercontent.com/dotnet/core-docs/master/docs/async/async-fsharp.md) contains the character.

Unassigned myself because it's unfortunately not just a matter of adding it back in.

### PR DESCRIPTION
**Pull request by cartermp**
- Add fixes for the F# doc headers to no longer have the "#" missing from the end.
- Add some grammar fixes and clearer words in the C#/VB doc

Fixes #157.

/cc @richlander @blackdwarf @mairaw

### CODE REVIEW

**Comment by mairaw** [comment]
Glad you found a fix for the # issue!

**Comment by cartermp** [comment]
Updated and rebased to reflect the changes in the C# async article.

File: docs/async/async-csharp-vb.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 By [Phillip Carter](https://github.com/cartermp)
 
-Async programming in C# and VB share a language-level asynchronous programming model which allows for easily writing asynchronous code without having to juggle callbacks or conform to a library which supports asynchrony. It follows what is known as the [Task-based Asynchronous Pattern (TAP)](https://msdn.microsoft.com/en-us/library/hh873175%28v=vs.110%29.aspx).
```
Comment (mairaw) [inline docs/async/async-csharp-vb.md]: We should change all the mentions to VB to Visual Basic, as we did with the titles.

Comment (cartermp) [inline docs/async/async-csharp-vb.md]: Good catch!

**Comment by cartermp** [comment]
Rebased and merging in so I can begin the changes to async docs as per #176.  Thanks for the feedback @mairaw and @matthewreily!

**Pull request merged**